### DISCUSSION TITLE
Custom Axes scales beyond what's bundled

### DISCUSSION THREAD

**Discussion by phobson** [vega/altair#3093] [Q&A]
Hey folks,

I've been getting deeper into Altair and I wasn't able to locate any examples or documentation about creating custom axes transformations and scales beyond the bundled [ScaleTypes](https://altair-viz.github.io/user_guide/generated/core/altair.ScaleType.html#altair.ScaleType). 

To be specific, I wrote a library called [mpl-probscale](https://matplotlib.org/mpl-probscale/) that implements probability scales in (wait for it...) matplotlib. 

Short version: probability scales take a quantile plot and convert the quantiles to probabilities (and back) based on the CDF/PPF of some user-specified probability distribution (typically normal or weibull in my field). 

Here's an example comparing three different probability scales and a linear scale all with the same tick locations:
![image](https://github.com/altair-viz/altair/assets/1163939/26038366-878c-4f3b-9382-92b06131142c)

(here's a link to the [docs page with this image](https://matplotlib.org/mpl-probscale/tutorial/getting_started.html#probabilty-scales), just scroll down a bit) 

I'd love to be able to do something like this in Altair, but I'm getting a little lost in the internals. Can anyone point me in the right direction?

**Comment by joelostblom** [comment]
Hmm I am not sure if there is a way to control where the ticks are placed beyond the different scale types that you already referred to. This might be worth asking on the VegaLite issue tracker (maybe after checking out if their more elaborate scale documentation is helpful, [eg quantile/quantizing scales](https://vega.github.io/vega-lite/docs/scale.html#quantile)?)

**Reply by phobson** [reply]
Thanks for the response, Joel. I posted a similar question in the vega-lite GH page. I'll update this discussion if I learn anything.
